sympy/solvers/solveset.py

```diff
@@ -1370,16 +1368,16 @@ def _solve_as_bivariate(lhs, rhs, symbol, domain):
 
 def _is_lambert(f, symbol):
```

Comment:
This should probably be renamed to `_possible_lambert` and docstring updated to say: "return True if any factor in `f` is a power, log or exp containing the given symbol". Note the change regarding symbol. See `_is_logarithmic` for an example of a better docstring for the function.

Also, `_term_factors` could get `*syms` and then `yield mul_arg if not syms or mul_args.has(*syms)`